support/k8s/k8s-workload-registrar/mode-crd/controllers/spiffeid_controller.go

```diff
@@ -212,15 +212,17 @@ func (r *SpiffeIDReconciler) updateOrCreateSpiffeID(ctx context.Context, spiffeI
 
 // deleteSpiffeID deletes the specified entry on the SPIRE Server
 func (r *SpiffeIDReconciler) deleteSpiffeID(ctx context.Context, spiffeID *spiffeidv1beta1.SpiffeID) error {
-	err := deleteRegistrationEntry(ctx, r.c.R, *spiffeID.Status.EntryId)
-	if err != nil {
-		return err
-	}
+	if spiffeID.Status.EntryId != nil {
+		err := deleteRegistrationEntry(ctx, r.c.R, *spiffeID.Status.EntryId)
+		if err != nil {
+			return err
+		}
 
-	r.c.Log.WithFields(logrus.Fields{
-		"entryID":  spiffeID.Status.EntryId,
-		"spiffeID": spiffeID.Spec.SpiffeId,
-	}).Info("Deleted entry")
+		r.c.Log.WithFields(logrus.Fields{
+			"entryID":  spiffeID.Status.EntryId,
```

Comment:
Need to dereference `EntryId`?
```suggestion
			"entryID":  *spiffeID.Status.EntryId,
```